### ISSUE TITLE
Dump structures

### ISSUE THREAD

**Issue by Kielbashnikov** [maruohon/tellme#41]
I would love to get full list of structures with their IDs, would be very useful with creating modpacks and such.

**Comment by maruohon** [comment]
There is a structure dump. I forget what the exact name is on Fabric vs. Forge, but if you use the `all` dump type then it's going to be one of them. Or you can check the code of the `SubCommandDump` class in the appropriate branch to see the dump names on your MC version and mod loader (which you didn't even mention).

**Comment by Kielbashnikov** [comment]
It's forge 1.20.1 with 0.9.0 tellme. I already dumped all and from what I can see it only dumps sctructure pieces, placements, pool elements and processors, I cannot see list of just structures avaible in the game. I might be blind, tell me if that is the case :D

**Comment by maruohon** [comment]
There should also be a `structure-type` dump.

**Comment by Kielbashnikov** [comment]
Yes, there is but it's quite empty, I mean, just few of all sctructures are mentioned (for example there are no villages mentioned).

**Comment by maruohon** [comment]
Can you post the output of that? I don't have a 1.20.1 Forge environment set up, nor the 1.20.1 update PR built.

Here is what I get in the Fabric mod version:
```
+----------------------------+
| Registry name              |
+----------------------------+
| minecraft:buried_treasure  |
| minecraft:desert_pyramid   |
| minecraft:end_city         |
| minecraft:fortress         |
| minecraft:igloo            |
| minecraft:jigsaw           |
| minecraft:jungle_temple    |
| minecraft:mineshaft        |
| minecraft:nether_fossil    |
| minecraft:ocean_monument   |
| minecraft:ocean_ruin       |
| minecraft:ruined_portal    |
| minecraft:shipwreck        |
| minecraft:stronghold       |
| minecraft:swamp_hut        |
| minecraft:woodland_mansion |
+----------------------------+
| Registry name              |
+----------------------------+
```

Indeed villages are not here (anymore?) for whatever reason, but the other vanilla structures seem to be there for the most part. Has some stuff been moved to work via the jigsaw blocks now somehow? I still haven't even looked up what and how those do exactly.

**Comment by Kielbashnikov** [comment]
This is my output (I have couple mods installed as you can see):

-------------
Registry name
-------------
betterdungeons:small_nether_dungeon
betterdungeons:spider_dungeon
dungeoncrawl:dungeon
dungeonnowloading:generic_jigsaw_structure
friendsandfoes:citadel_structure
friendsandfoes:iceologer_cabin_structure
friendsandfoes:illusioner_shack_structure
friendsandfoes:illusioner_training_grounds_structure
lithostitched:jigsaw
minecraft:buried_treasure
minecraft:desert_pyramid
minecraft:end_city
minecraft:fortress
minecraft:igloo
minecraft:jigsaw
minecraft:jungle_temple
minecraft:mineshaft
minecraft:nether_fossil
minecraft:ocean_monument
minecraft:ocean_ruin
minecraft:ruined_portal
minecraft:shipwreck
minecraft:stronghold
minecraft:swamp_hut
minecraft:woodland_mansion
supplementaries:way_sign
tlc:lost_castle
villagesandpillages:village_witch_structure
yungsapi:yung_jigsaw
-------------
Registry name
-------------
I do not know what that jigsaw is. Just to explain: I have installed sparse structures and I wanted to set diffrent spreads for structures, but without a list it would be pain in the ass to write down every strtucture id by hand (funny part is that sparse structures has dump structures command, but it doesn't work).

**Comment by maruohon** [comment]
Well if they are not in the vanilla registry then TellMe can't dump them either, without adding explicit direct support for that mod. And in general I don't want to add direct dependencies to other mods, as that is a lot of extra work and maintenance overhead to maintain those.

**Comment by OarisHunter** [comment]
could this be investigated more potentially? The jigsaw entry from vanilla can generate pillager outposts, villages, bastion remnants, ruined portals, and ancient cities (which explains why they aren't in the list). Modded ones I'm sure are also used to generate several types of structures. Maybe that data is part of the registry but not truly associated with structures? I couldn't say, but in cases where structure ids are needed, grouping all structures into a single jigsaw entry doesn't really accomplish anything useful.

Minecraft can locate structures via the /locate command, that info has to come from somewhere no?

**Comment by Kawodeda** [comment]
I've encountered same issue. My modpack has quite an amount of structures that are not present in tellme dump, although i can locate them via minecraft /locate
